homeassistant/components/huawei_lte/__init__.py

```diff
@@ -612,6 +555,18 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
         data[CONF_MAC] = []
         hass.config_entries.async_update_entry(config_entry, data=data)
         _LOGGER.info("Migrated config entry to version %d", config_entry.version)
+    if config_entry.version == 3:
```

Comment:
Is this migration really needed. Those superfluous keys seem harmless?

Reply:
I don't like leaving old cruft behind, it only serves to potentially confuse.

Reply:
It also breaks downgrading capabilities, which will be confusing for some as well.

Reply:
Quicker without it.

Reply:
Done, replaced with a related comment.